client/js/ifml/fromjson/intermediate/index.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2018, the IFMLEdit.org project authors. Please see the
+// AUTHORS file for details. All rights reserved. Use of this source code is
+// governed by a MIT-style license that can be found in the LICENSE file.
+/*jslint node: true, nomen: true */
+"use strict";
+
+var elementRules = require('./elementrules'),
+    relationRules = require('./relationrules'),
+    extender = require('../extender'),
+    core = require('almost-core'),
+    createTransformer = require('almost').createTransformer;
+
+var transformer = createTransformer(
+    {
+        element: elementRules,
+        relation: relationRules
+    },
+    core.merge(
+        core.none(),
+        {
+            elements: core.reduceBy('id', core.merge(core.mergeOrSingle(), {
```

Comment:
If the rules generate directly elements you can avoid the external _merge_ and just use the internal _reduceBy_